### PR TITLE
chore: bump node-gptscript to v0.9.1 

### PR DESCRIPTION
**Pull request by njhale**
Bump the version of node-gptscript and refactor to account for breaking changes.

### CODE REVIEW

**Review by thedadams — APPROVED** [review]

File: server/api/story/index.post.ts [hunk 1]
```diff
@@ -1,49 +1,102 @@
...
+    console.log(`generating story for: ${JSON.stringify(request)}`)
 
     // Run the script with the given prompt and number of pages
     const {storiesVolumePath, scriptPath, gptscriptCachePath} = useRuntimeConfig()
-    const opts: { cacheDir?: string } = gptscriptCachePath ? { cacheDir: gptscriptCachePath } : {}
-    const {stdout, stderr, promise} = await gptscript.streamExecFile(
-        `${scriptPath}`, `--story ${request.prompt} --pages ${request.pages} --path ${storiesVolumePath}`, opts)
+    const opts = {
+        input: `--story ${request.prompt} --pages ${request.pages} --path ${storiesVolumePath}`,
+        credentialOverrides: ['sys.openai:OPENAI_API_KEY'],
```
Comment (tylerslaton) [inline server/api/story/index.post.ts:35]: Is this necessary? I haven't seen this before.

**Review by tylerslaton — COMMENTED** [review]
Does this break progress? From what I can tell we no longer push updates when progress happens, just on CallFinish and Text being processed.

**Review by tylerslaton — APPROVED** [review]
Nevermind, I re-read the code and everything looks good here.

File: server/api/story/index.post.ts (hunk 1 above)
Comment (njhale) [reply server/api/story/index.post.ts:35]: yes, we're not surfacing prompts, so it's necessary to override with the environment variables.

**Pull request merged**